packages/webpack-configs/src/build.ts

```diff
@@ -0,0 +1,166 @@
+import type { Config as SwcConfig } from "@swc/core";
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import MiniCssExtractPlugin from "mini-css-extract-plugin";
+import { createRequire } from "node:module";
+import path from "path";
+import TerserPlugin from "terser-webpack-plugin";
+import type { Configuration as WebpackConfig } from "webpack";
+import { applyTransformers, type WebpackConfigTransformer } from "./transformers/applyTransformers.ts";
+
+// Using node:module.createRequire until
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta/resolve
+// is available
+const require = createRequire(import.meta.url);
+
+type MiniCssExtractPluginOptions = NonNullable<ConstructorParameters<typeof MiniCssExtractPlugin>[number]>;
+
+export function defineBuildHtmlWebpackPluginConfig(options: HtmlWebpackPlugin.Options = {}) {
+    const {
+        template = path.resolve("./public/index.html"),
+        ...rest
+    } = options;
+
+    return {
+        ...rest,
+        template
+    } as HtmlWebpackPlugin.Options;
```

Comment:
instead of casting after like this, you should specify a return value for the function instead

Reply:
Thanks, fixed it for other functions too.